test/uniswapTrades.test.ts

```diff
@@ -595,4 +614,157 @@ describe('Uniswap', () => {
+        CurrencyAmount.fromRawAmount(WSTETH, inputWSTETH),
+        TradeType.EXACT_INPUT
+      )
+
+      const methodParameters = SwapRouter.swapCallParameters([
+        wrapSTETH,
+        new UniswapTrade(buildTrade([trade]), swapOptions({ payerIsRouter: true })),
+      ])
+      registerFixture('_UNISWAP_V3_001_STETH_FOR_WETH', methodParameters)
+      expect(hexToDecimalString(methodParameters.value)).to.eq('0')
+      // other assertions carried out in forge
+    })
+
+    it('encodes a single exactInput STETH -> WSTETH -> ETH swap with Permit', async () => {
+      const inputSTETH = expandTo18DecimalsBN('0.001')
+      const inputWSTETH = await getWStethPerSteth(inputSTETH.sub(1))
+      const permit2Data = makePermit(STETH_ADDRESS(1), inputSTETH.toString(), undefined, FORGE_ROUTER_ADDRESS)
+      const signature = await generatePermitSignature(permit2Data, wallet, 1, FORGE_PERMIT2_ADDRESS)
+
+      const WrapSTETHPermitData = {
+        ...permit2Data,
+        signature,
+      }
+
+      const wrapSTETH = new WrapSTETH(inputSTETH, 1, WrapSTETHPermitData)
+
+      const trade = await V3Trade.fromRoute(
+        new RouteV3([WETH_WSTETH_V3], WSTETH, ETHER),
+        CurrencyAmount.fromRawAmount(WSTETH, inputWSTETH),
+        TradeType.EXACT_INPUT
+      )
+
+      const methodParameters = SwapRouter.swapCallParameters([
+        wrapSTETH,
+        new UniswapTrade(buildTrade([trade]), swapOptions({ payerIsRouter: true })),
+      ])
+      registerFixture('_UNISWAP_V3_001_STETH_FOR_ETH', methodParameters)
+      expect(hexToDecimalString(methodParameters.value)).to.eq('0')
+      // other assertions carried out in forge
+    })
+
+    it('encodes a single exactInput STETH -> WSTETH -> WETH exactOutput swap', async () => {
+      const outputWETH = expandTo18DecimalsBN('0.001')
+
+      // Trade Configurations
+      const swapOpts = swapOptions({ payerIsRouter: true })
+      const trade = await V3Trade.fromRoute(
+        new RouteV3([WETH_WSTETH_V3], WSTETH, WETH),
+        CurrencyAmount.fromRawAmount(WETH, outputWETH),
+        TradeType.EXACT_OUTPUT
+      )
+
+      // Wrap Configurations
+      const maximumWstethIn = BigNumber.from(trade.maximumAmountIn(swapOpts.slippageTolerance).quotient.toString())
+      const inputSTETH = await getStethPerWsteth(maximumWstethIn.add('1'))
+      const permit2Data = makePermit(STETH_ADDRESS(1), inputSTETH.toString(), undefined, FORGE_ROUTER_ADDRESS)
+      const signature = await generatePermitSignature(permit2Data, wallet, 1, FORGE_PERMIT2_ADDRESS)
+      const WrapSTETHPermitData = {
+        ...permit2Data,
+        signature,
+      }
+      const wrapSTETH = new WrapSTETH(inputSTETH, 1, WrapSTETHPermitData)
+      const unwrapSTETH = new UnwrapSTETH(SENDER_AS_RECIPIENT, 0, 1)
+
+      const methodParameters = SwapRouter.swapCallParameters([
+        wrapSTETH,
+        new UniswapTrade(buildTrade([trade]), swapOpts),
+        unwrapSTETH,
```

Comment:
do you need an unwrap here for STETH? since its WETH out

Reply:
yes, bc it's exactOut so the input amount is uncertain